resdk/resources/base.py

```diff
@@ -260,3 +274,16 @@ def __repr__(self):
         return "{} <id: {}, slug: '{}', name: '{}'>".format(
             self.__class__.__name__, self.id, self.slug, self.name
         )
+
+    def _resource_setter(self, payload, resource, field):
+        """Set ``resource`` with ``payload`` on ``field``."""
+        if isinstance(payload, resource):
+            setattr(self, field, payload)
+        elif isinstance(payload, dict):
+            setattr(self, field, resource(resolwe=self.resolwe, **payload))
+        elif isinstance(payload, int):
+            setattr(self, field, resource(resolwe=self.resolwe, id=payload))
```

Comment:
Populate the resource object.

Reply:
Please check if you like the solution now.